### PR TITLE
Add tags and related actions and filters to SQS

### PR DESCRIPTION
**Pull request by salwandhruvdev**
This is related to issue #1679 
Work done:
- Added tags to augment 
- Added "mark-for-op", "tag", "remove-tag"
- Added tests for the three actions
- Refactored old test cases
- Added marked-for-op filter

Thanks.

### CODE REVIEW

File: c7n/resources/sqs.py [hunk 1]
```diff
@@ -60,13 +64,20 @@ def _augment(r):
...
                 if e.response['Error']['Code'] == 'AccessDenied':
                     self.log.warning("Denied access to sqs %s" % r)
                     return
                 raise
+            tag_list = []
+            for k, v in tag_dict.items():
+                tag_list.append({'Key': k, 'Value': v})
 
             queue['QueueUrl'] = r
+            queue['Tags'] = tag_list or []
```
Comment (kapilt) [inline c7n/resources/sqs.py]: the or seems weird here, it always at min an empty list

File: tests/test_sqs.py [hunk 2]
```diff
@@ -176,8 +176,86 @@ def test_sqs_remove_named(self):
...
+            'resource': 'sqs',
+            'filters': [{'QueueUrl': queue_url}],
+            'actions': [
+                {'type': 'remove-tag',
+                 'tags': ['remove-this-tag']}]},
+            session_factory=session_factory)
+
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        tags_afer_run = client.list_queue_tags(
```
Comment (kapilt) [inline tests/test_sqs.py]: s/afer/after

**Comment by salwandhruvdev** [comment]
Ready for review. The build is failing because it needs updated botocore to pick up `list_tags_for_queue`. @kapilt

**Comment by kapilt** [comment]
its still failing but not because of dependency issues, but code issues. do the tests run locally?

**Comment by kapilt** [comment]
we're working on some build config changes that should hopefully alleviate this.

**Comment by kapilt** [comment]
fwd reference on ci library fixes. https://github.com/tox-dev/tox/pull/668

File: .drone.yml [hunk 3]
```diff
@@ -3,6 +3,7 @@ build:
   unit_tests:
     image: python:2-slim
     commands:
+      - rm -rf .tox
```
Comment (kapilt) [inline .drone.yml]: okay for getting ci green, which to be noted this is not at yet, but not a desirable solution

Comment (kapilt) [reply .drone.yml]: wondering if we should just go ahead and do what botocore does and separate the install to a different script, my fix for tox seems to be running afoul of tests that are dependent on the extant behavior.

Comment (jhnlsn) [reply .drone.yml]: The same does not appear to work on travis.

Comment (kapilt) [reply .drone.yml]: travis has its own config file.

**Pull request merged**